---
title: "Final Report"
author: "Ana Fernandes PG38284, João Capela PG38274, Tiago Ferreira PG33765, Tiago Oliveira PG24096"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    collapsed: false
    depth: 3
    number_sections: true 
    theme: flatly 
    highlight: tango
    df_print: paged
    code_download: true
    code_folding: show
---


```{r, include=FALSE}
BiocManager::install(c("hgu95av2.db","topGO","genefilter","DESeq2"))
library(topGO)
library(genefilter)
library(hgu95av2.db)
library(DESeq2)
``` 


# Introduction
  Gastric Cancer (GC) is recognized as one of the causes of cancer incidence around the globe. Moreover, it was identified as the third major cause of death by oncological disease. To the best of our knowledge, there are several factors associated to the formation and progression of GC, namely molecular, genetic, environmental factors, and epigenetic alterations. 
  As for epigenetic alterations, DNA methylations, histone modifications, nucleossome positioning and non codifying RNA. Following up the work of Baek and collaborators (2015), this analysis will focus on DNA methylation. This phenomenon has been reported as one of the major causes of carcinogenesis.
  Taking into account the work of Baek and collaborators (2015), the main goal of present work is to analyze the RNA-seq data obtained from GC patients and healthy samples. Moreover, it aims at finding differences between proximal and distal regulatory regions of GC patient samples and healthy cases. For this, RNA-seq data of normal and tumoral tissues was analyzed and characterized. This data was divided into the following groups: \textit(purified gastric cancer cell (sc), cultured gastric cancer cell (dc) e mixture of several gastric cancer cell clones (mc)) of three patients (csc1, csc2, csc3)  and two replicates (csc1_sc2, csc1_sc3) (Table 1)[1].
  

![Table 1 - data organization.](table1.png)



# Methods

The study data is constituted by 15 libraries of RNA-Seq paired-end, which were prepared by PCR using the TruSeq RNA Sample Prep Kit (Illumina, San Diego, CA, USA) and sequenced using Hiseq-2000 (Illumina). Each library is in “.fastq.gz” format.

## Data quality control

FastQC (version: 0.11.8) was used in order to generate a quality control report of each library. Then, these reports were exported for the folder “fastQC_reports/”, using the following command:

```{bash, eval=FALSE}
fastqc -o fastQC_reports/ SRR*
```

## Reads management

The library reads were filtered by quality and the adaptors sequences were cut with Trimmomatic (version: 0.39), using the following command:


```{bash, eval=FALSE}
for i in {5..8}
do 
	trimmomatic PE SRR104158$i\_1.fastq.gz SRR104158$i\_2.fastq.gz Trim/SRR104158$i\_1P Trim/SRR104158$i\_1U Trim/SRR104158$i\_2P Trim/SRR104158$i\_2U ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2:keepBothReads LEADING:0 TRAILING:0 SLIDINGWINDOW:0:0
done

for i in {22..32}
do 
	trimmomatic PE SRR8469$i\_1.fastq.gz SRR8469$i\_2.fastq.gz Trim/SRR8469$i\_1P Trim/SRR8469$i\_1U Trim/SRR8469$i\_2P Trim/SRR8469$i\_2U ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2:keepBothReads LEADING:0 TRAILING:0 SLIDINGWINDOW:0:0 
done
```


No reads quality parameter was inserted, so that Trimmomatic could determine which limits of the reads quality should be inserted, adequately, to each library.

In order to remove the adaptors sequences, the sequences used in the PCRs were indicated. Moreover, the mismatch and score parameters were set. It was also indicated that the Trimmomatic should consider any adaptor with a given minimum size (2). Lastly, the reverse reads were considered as well. 

As for the FastQC report results, no filter was considered per level of quality.

After the Trimmoatic operations, the results were thoroughly analysed, so that the quality could be checked and possible errors identified. For that, that following command was run.


````{bash, eval=FALSE}
fastqc -o fastQC_reports/ Trim/SRR*"P"
````



## Alignment

The third chromossome of the GRCh38 assembly of the human genome was used as a reference. This reference was indexed and used for the Burrows-Wheeler Aligner (BWA)  (version:0.7.17-r1188) library alignment.

The following commands were used for Indexation, aligment and creation of the final ".bam" files. 


````{bash, eval=FALSE}
bwa index Index/Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz  # Indexacao

ls *.fastq.gz | cut -d "_" -f1 | uniq > fastq_names.txt # Lista de amostras


for fq in $(cat fastq_names.txt)
do
	bwa aln Index/Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz Trim/$fq"_1P" > aln_ch3/$fq"_1.sai"
	bwa aln Index/Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz Trim/$fq"_2P" > aln_ch3/$fq"_2.sai"
done


for fq in $(cat fastq_names.txt)
do
	bwa sampe Index/Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz aln_ch3/$fq"_1.sai" aln_ch3/$fq"_2.sai" Trim/$fq"_1P" Trim/$fq"_2P" | samtools view -S -b > bams_chr3/$fq".bam"
done
````

## Countings

For the gene counting, a .gtf format file of the human genome's GRCh38 assembly was used. The features were counted using the HTSeq-count (version:0.11.2)


````{bash, eval=FALSE}
for fq in $(cat fastq_names.txt)
do
	htseq-count -f bam -s no bams_chr3/$fq".bam" gff/Homo_sapiens.GRCh38.96.gtf.gz > readcounts/$fq"_readcounts"
done

echo "geneID csc1_sc2 csc1_dc2 csc1_sc3 csc1_dc3 csc1-dc csc1-mc csc1-sc csc2-dc csc2-mc csc2-sc csc3-dc csc3-mc csc3-sc normal1 normal2" > reads.tab
paste *_readcounts | cut -f1,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30 >> reads.tab

grep "^ENS" reads.tab > reads_final.tab

````

The countings were normalized as Reads Per Kilobase Million (RPKM). The number of reads in the sample was determined using Samtools.


```{bash, eval=FALSE}
for fq in $(cat fastq_names.txt)
do
  samtools view -c bams_chr3/$fq".bam"
```

The RPKM was calculated for each feature using the following python code:

```{python, eval=FALSE}
import csv
if __name__ == '__main__':
	dic_features = {}
	with open("gff/Homo_sapiens.GRCh38.96.gtf", "r") as gtf:
		for line in gtf:
			if not line[0].startswith("#"):
				line = line.split("\t")
				id = line[8].split(" ")[1].strip(";")[1:-1]
				dic_features[id] = int(line[4]) - int(line[3])


	with open("readcounts/reads_final.tab", "r") as read_counts:
		with open("readcounts/rpkm_reads.tab", "w+") as rpkm_counts:
			table = []
			total_reads = [76631818, 73488960, 77161352, 77611646, 58909066,
						   79698582, 68670218, 77929308, 56466226, 43113480,
						   38844902, 61500262, 64124276, 29181404, 42569474]
			for line in read_counts:
				line = line.split("\t")
				if line[0] in dic_features.keys():
					if dic_features[line[0]] != 0:
						for i in range(1,16):
							if line[1] != 0:
								line[i] = (10**6*int(line[i])/(total_reads[i-1]*dic_features[line[0]]))
				line[15] = int(str(line[15]).strip("\n"))
				table.append(line)
			cw = csv.writer(rpkm_counts)
			cw.writerows(table)
```


# Data Analysis

## Reads Quality reports

The resultant reports from the fastQC, demonstrate that the initial libraries do not contain adaptors. All the reads possessed between 62 and 79 base pairs and good quaity in all of the metrics.
Even though the report has indicated, in the "Per base sequence content" and "Sequence Duplication Levels" fields, that some problems could be pointed, by analysing thouroghly the reports, we concluded that these results were expected in RNA-Seq data. 

After the usage of the Trimmomatic, some libraries have presented quality values lower than the original ones. 


## Loading the data and a DeSeqDataSet object

The countings were imported and analysed using DeSeq2 (version: 1.22.2):


```{r}
reads <- read.table("reads_final.tab", h=T, row.names=1, sep = "\t")
```

Then some conditions were set to compare the data.

```{r}
metadata<-read.table("metadata.txt", sep = ";")
colnames(metadata)<-c("ID","sample","disease")
colnames(reads)=c("csc1_sc2","csc1_dc2","csc1_sc3","csc1_dc3", "csc1-dc","csc1-mc","csc1-sc", "csc2-dc","csc2-mc", "csc2-sc", "csc3-dc","csc3-mc", "csc3-sc", "normal1", "normal2")
```

Before running the DeSeq functions, the genes with the lowest counts were filtered.


```{r}
reads <- reads[ rowSums(reads) > 1, ]
```

## Diferential Expression Analysis


```{r}
dds <- DESeqDataSetFromMatrix(countData = reads,colData = metadata, design = ~ disease)
dds <- DESeq(dds)
# Expressão diferencial de amostras que pertencem a CI vs C2
res <- results(dds)
res
# Oraganização de tabelas por ordem do menor valor-p
resOrdered <- res[order(res$padj),]
# Sumarização de dados 
summary(res)
# Tem-se muitos valores com p-value inferior a 0.1?
sum(res$padj < 0.1, na.rm=TRUE)
```


## Results exploration

```{r}
# MA-plot
plotMA(res, main="DESeq2", ylim=c(-2,2))
```

As for the read counts, the MA plot can represent a global vision of the differences between the samples. In this graph, it is easy to check that there are various subexpressed and overexpressed genes. 


```{r}
# representar graficamente os read-counts para o gene mais sigfnificante
plotCounts(dds, gene=which.min(res$padj), intgroup="disease")
# Exportaçao para um .CSV
write.csv(as.data.frame(resOrdered),file="disease_treated_results.csv")
```

## Data transformation and visualization

Counting transformation:
In order to test differential expression, discrete distributions were used for the "raw counts".

```{r}
# VST: varianceStabilizingTransformation
vsd <- varianceStabilizingTransformation(dds, blind=FALSE)
head(assay(vsd), 20)
# comparar o comando previo com head(counts(dds), 3)
# Heatmap da matriz dos counts
select <- rownames(head(resOrdered,20))
vsd.counts <- assay(vsd)[select,]
colnames(vsd.counts)<-metadata$disease
df <- as.data.frame(colData(dds)[,c("disease")])
# manter a ordenação das linhas
library(pheatmap)
pheatmap(vsd.counts, cluster_rows=FALSE,show_colnames = T)
# cluster por linha e coluna
pheatmap(vsd.counts)
```

The results from the differential expression analysis revealed that the gene with the ENSG00000269028 ENSEMBLE ID was overexpressed for the cancer cells. The gene is MTRNR2L12 and the codified protein is dubbed as \textit(Humanin-like 12). This protein is codified by a small Open Reading Frame (ORF) in the mitochondrial DNA (mtDNA) [3]. Its function is to supress apoptosis [2]. Once that an apoptosis disruption is directly linked with cancer [4], this is the expected result.

## Heatmaps about the Eucledian distance between samples


```{r}
# calcular a distancia entre amostras
sampleDists <- dist(t(assay(vsd)))

# preparar a matriz de distancias
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- dds$sample
colnames(sampleDistMatrix) <- NULL
```

```{r}
library(RColorBrewer)
colors <- colorRampPalette( rev(brewer.pal(9, "RdBu")) )(255)
pheatmap(sampleDistMatrix, clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists, col=colors)
```

In the above heatmap, it is possible to verify that the normal samples data tends to group, as well as the other samples. However, the 2 and 3 replicates, collected from the first patient, group themselves, as well as the rest generate another cluster.

## Principal Component Analysis

```{r}
plotPCA(vsd, intgroup=c("disease"))

```


The Principal Component Analysis (PCA) revealed a natural separation between the healthy people samples and the other from the Gastric Cancer patients.

In order to annotate the relevant genes, the following command was used:

```{r}
anno <- AnnotationDbi::select(hgu95av2.db, 
                              keys=rownames(resOrdered), 
                              columns=c("SYMBOL","SYMBOL", "GENENAME"),
                              keytype="ENSEMBL")

```


# Discussion


The results from the RNA-Seq data analysis did not allow to take relevant conclusions, even though the results correponded with the expected.
In order to come across with better conclusions, a \textit(genome-wide analysis) must be performed. Furthermore, an analysis such as the mentioned would give us importatant insights into the understanding of how epigenomic alterations in proximal and distal regions of the human genome could have an importatn role in the development of GC.


[1]Baek, S., Kim, M., Bae, D., Kim, J., Kim, H., Han, M., Oh, S., Kim, Y. and Kim, S. (2015). Integrated epigenomic analyses of enhancer as well as promoter regions in gastric cancer. Oncotarget

[2]Guo B, Zhai D, Cabezas E, Welsh K, Nouraini S, Satterthwait AC, Reed JC (May 2003). "Humanin peptide suppresses apoptosis by interfering with Bax activation". Nature. 423 (6938)

[3]Bodzioch M, Lapicka-Bodzioch K, Zapala B, Kamysz W, Kiec-Wilk B, Dembinska-Kiec A (October 2009). "Evidence for potential functionality of nuclearly-encoded humanin isoforms". Genomics. 94 (4): 247-56. doi:10.1016/j.ygeno.2009.05.006. PMID 19477263.

[4]Scott W. Lowe, Athena W. Lin, Apoptosis in cancer , Carcinogenesis, Volume 21, Issue 3, March 2000, Pages 485-495
